File: tauri/src-tauri/src/core/prompt/checklist.rs
use super::PromptPart;

pub struct ChecklistBehaviorPart;

impl PromptPart for ChecklistBehaviorPart {
    fn render(&self) -> String {
        "## PROGRESS TRACKING\n\n\
         For complex tasks, maintain a checklist using the `manage_todos` tool:\n\
         - `[ ]` pending  `[>]` in-progress  `[x]` completed\n\
         - Format: `[status] Task Name <- context`\n\n\
         Update the checklist as you make progress. The user sees it in real-time."
            .to_string()
    }
}

File: electron/electron/core/prompt/checklist.ts
import type { PromptPart } from './types'

class ChecklistBehaviorPart implements PromptPart {
    render(): string {
        return `## PROGRESS TRACKING

For complex tasks, maintain a checklist using the \`manage_todos\` tool:
- \`[ ]\` pending  \`[>]\` in-progress  \`[x]\` completed
- Format: \`[status] Task Name <- context\`

Update the checklist as you make progress. The user sees it in real-time.`
    }
}

export default ChecklistBehaviorPart
